src/indexables-page.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Yoast\WP\Test_Helper;
+
+use Yoast\WP\SEO\Helpers\Indexables_Page_Helper;
+
+/**
+ * Class to manage registering and rendering the admin page in WordPress.
+ */
+class Indexables_Page implements Integration {
+
+	/**
+	 * Holds our option instance.
+	 *
+	 * @var Option
+	 */
+	private $option;
+
+	/**
+	 * Class constructor.
+	 *
+	 * @param Option $option Our option array.
+	 */
+	public function __construct( Option $option ) {
+		$this->option = $option;
+	}
+
+	/**
+	 * Adds the required hooks for this class.
+	 */
+	public function add_hooks() {
+		\add_filter( 'wpseo_posts_threshold', [ $this, 'indexables_posts_threshold' ], 10, 1 );
+		\add_filter( 'wpseo_analyzed_posts_threshold', [ $this, 'indexables_analyzed_posts_threshold' ], 10, 1 );
+
+		\add_action( 'admin_post_yoast_seo_test_indexables_page', [ $this, 'handle_submit' ] );
+	}
+
+	/**
+	 * Filter minimum post threshold.
+	 *
+	 * @param int $number The current post threshold.
+	 *
+	 * @return int The current post threshold.
+	 */
+	public function indexables_posts_threshold( $number ) {
+		if ( $this->option->get( 'indexables_posts_threshold' ) > 0 ) {
+			return $this->option->get( 'indexables_posts_threshold' );
+		}
+
+		return $number;
+	}
+
+	/**
+	 * Filter minimum analyzed post threshold.
+	 *
+	 * @param int $number The current analyzed post threshold.
+	 *
+	 * @return int The current analyzed post threshold.
+	 */
+	public function indexables_analyzed_posts_threshold( $number ) {
+		if ( $this->option->get( 'indexables_analyzed_posts_threshold' ) > 0 ) {
+			return $this->option->get( 'indexables_analyzed_posts_threshold' );
+		}
+
+		return $number;
+	}
+
+	/**
+	 * Retrieves the controls.
+	 *
+	 * @return string The HTML to use to render the controls.
+	 */
+	public function get_controls() {
+		// phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound -- Using WPSEO hook.
+		$placeholder_thresholds = \apply_filters( 'wpseo_posts_threshold', Indexables_Page_Helper::POSTS_THRESHOLD );
```

Comment:
Lets check if that class exists before we call the `Indexables_Page_Helper::POSTS_THRESHOLD` :) 

In fact, lets see if we can wrap all added functionality under that check, so that we don't display anything if the user eg. is on a not-upgraded Yoast Free version (probably in the plugin.php file?)